core/src/apps/cardano/sign_tx/signer.py

```diff
@@ -0,0 +1,903 @@
+from micropython import const
+from typing import TYPE_CHECKING
+
+from trezor import messages, wire
+from trezor.crypto import hashlib
+from trezor.crypto.curve import ed25519
+from trezor.enums import (
+    CardanoAddressType,
+    CardanoCertificateType,
+    CardanoTxWitnessType,
+)
+
+from apps.common import cbor, safety_checks
+
+from .. import layout
+from ..address import (
+    ADDRESS_TYPES_PAYMENT_SCRIPT,
+    derive_address_bytes,
+    derive_human_readable_address,
+    get_address_bytes_unsafe,
+    get_address_type,
+    validate_output_address,
+    validate_output_address_parameters,
+)
+from ..auxiliary_data import (
+    get_auxiliary_data_hash_and_supplement,
+    show_auxiliary_data,
+    validate_auxiliary_data,
+)
+from ..certificates import (
```

Comment:
I think it might be a good idea, but we need to be comfortable with

1. making some identifier even longer, e.g. `address.validate_output_address_parameters` instead of `validate_output_address_parameters`, and
2. making some function definitions a bit harder to parse, i.e. we would have `derive_bytes` - bytes of what?

Reply:
Well, using module imports is a good idea for sure (even the original refactor proposal states that "we should just import layout and call the functions as layout.something, **dtto for a lot of other things**". I will add the import change and name shortening as separate commits (so that it's clear what we are suggesting) and we'll see what Trezor devs say.